### PR TITLE
Update deprecated clock source settings

### PR DESCRIPTION
**Pull request by m-gorecki**
XTAL_32768, XTAL_RC, XTAL_32786_SYNTH are now deprecated and we should use MCU_LFCLK_SOURCE setting instead.

### CODE REVIEW

**Review by sjanc — APPROVED** [review]

File: hw/mcu/nordic/nrf51xxx/syscfg.yml [hunk 1]
```diff
@@ -97,32 +97,49 @@ syscfg.defs:
...
+
     XTAL_32768:
-        description: 'External 32k oscillator LF clock source.'
+        description: Use MCU_LFCLK_SOURCE instead
         value: 0
         restrictions:
-            - "!XTAL_32768_SYNTH"
             - "!XTAL_RC"
-
+            - "!XTAL_32768_SYNTH"
```
Comment (kasjer) [inline hw/mcu/nordic/nrf51xxx/syscfg.yml:116]: Why reverse line order?

File: hw/mcu/nordic/nrf51xxx/syscfg.yml [hunk 2]
```diff
@@ -97,32 +97,49 @@ syscfg.defs:
...
+        choices:
+            - LFRC      # 32.768 kHz RC oscillator
+            - LFXO      # 32.768 kHz crystal oscillator
+            - LFSYNTH   # 32.768 kHz synthesized from HFCLK
+
+    # Deprecated settings
+
     XTAL_32768:
-        description: 'External 32k oscillator LF clock source.'
+        description: Use MCU_LFCLK_SOURCE instead
```
Comment (kasjer) [inline hw/mcu/nordic/nrf51xxx/syscfg.yml:112]: I would be more specific in description:
```yml
description: Use "MCU_LFCLK_SOURCE: LXFO" instead
```

**Review by kasjer — APPROVED** [review]

File: hw/bsp/hifive1/syscfg.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@
 #
 
 syscfg.vals:
-    XTAL_32768: 1
+    MCU_LFCLK_SOURCE: LFXO
```
Comment (kasjer) [inline hw/bsp/hifive1/syscfg.yml:21]: This is not nordic chip. This setting does not apply for hifive1

**Review by kasjer — CHANGES_REQUESTED** [review]

**Review by kasjer — APPROVED** [review]
LGTM

**Pull request merged**